### DISCUSSION TITLE
Give the option in mermaid diagram to show only diagram

### DISCUSSION THREAD

**Discussion by jinjuewei** [GitbookIO/community#142] [Feature Requests]
### What's your feature/idea?

The mermaid plugin shows both diagram codes and the diagram. Can we have the option to show only the diagram?

### What's your desired outcome?

User can choose to show both the mermaid diagram codes and diagram, or only one of it.

### What's the impact of this for you?

- [ ] Critical
- [ ] High
- [X] Medium
- [ ] Low

**Comment by ghost** [comment]
The inclusion of both the mermaid code and the chart is bizarre to me. I concur that some sort of switch be provided.

**Comment by elizaodie** [comment]
I don't mind having the code visible in the editor, but on the published docs, I think it's distracting from the visual. Would be nice to be able to disable it.

**Comment by jeffabailey** [comment]
As a new user, this behavior is confusing.

It may be useful to allow the code to show, but by default it makes more sense to simply render the diagram.

**Comment by rachellawson** [comment]
Definitely weird seeing the code on the _published_ pages

**Comment by taranvohra** [comment]
Hey 👋, So we checked and there indeed was a regression where the Mermaid code was showing at all times on published content or when edit mode was locked. The fix has been deployed. Hope it solves your issues
